### PR TITLE
Add responses to ruby objects diagnostic

### PR DESCRIPTION
**Pull request by lisamsmith100**
FoF: 3.5, 3.5
I used solution branches of lessons and pry for responses.  need more
time.

### CODE REVIEW

**Comment by lisamsmith100** [comment]
I'd like to spend more time with mixins.

File: lib/diagnostic.rb [hunk 1]
```diff
@@ -26,7 +26,29 @@
...
+
+  def age
+    @age
+  end
+
+  def location=(location)
+    @surname = surname
+  end
+end
+
```
Comment (Jcornmanhomonoff) [inline lib/diagnostic.rb:51]: Correct, but 
```rb
def name
    @name
  end

  def age
    @age
  end

  def location=(location)
    @surname = surname
  end
```
 is unnecessary.

File: lib/diagnostic.rb [hunk 2]
```diff
@@ -35,7 +57,7 @@
 # "Somerville". Finally, assign the modified Person to `Response.dave` below.
 
 ## replace nil with your response, then continue your work on the next line
-dave = nil
+dave = Person.new("Dave", 32, "Ohio")
```
Comment (Jcornmanhomonoff) [inline lib/diagnostic.rb:60]: `dave.location = 'Somerville'`

File: lib/diagnostic.rb [hunk 3]
```diff
@@ -45,7 +67,20 @@
...
+
+  def solve_problems
+    puts string = 'think think think'
+  end
+end
+
+# define Class Ford
+class Ford
+end
+
```
Comment (Jcornmanhomonoff) [inline lib/diagnostic.rb:83]: You do not need to initialize again since Developer is inheriting from Person.

File: lib/diagnostic.rb [hunk 4]
```diff
@@ -105,7 +144,10 @@ def eat_meat(food)
 end
 
 ##
-# your response here
+require_relative 'Cat'
+require_relative 'Carnivorous'
+class Lion < Carnivorous
+  def
```
Comment (Jcornmanhomonoff) [inline lib/diagnostic.rb:150]:
```rb
class Lion < Cat
  include Carnivorous
  def roar
    puts 'ROAR!'
  end

```

File: lib/diagnostic.rb [hunk 5]
```diff
@@ -105,7 +144,10 @@ def eat_meat(food)
...
 ##
 
 # #
```
Comment (Jcornmanhomonoff) [inline lib/diagnostic.rb:153]:
```rb
class ComboAttack
  attr_reader :moves, :damage

  def initialize
    @moves = []
    @damage = 0
  end

  def punch
    @moves << 'punch'
    @damage += 5
    @damage *= multiplier
    self
  end

  def move(direction)
    @moves << "move #{direction}"
    self
  end

  def kick
    @moves << 'kick'
    @damage += 10
    @damage *= multiplier
    self
  end

  def self.possible_moves
    'kick, move, punch'
  end

  private

  def multiplier
    case moves
    when %w(punch move left kick)
      1.5
    when %w(kick punch up)
      2
    else
      1
    end
  end
end
```

**Comment by Jcornmanhomonoff** [comment]
4/8! Not bad! 🦄

**Pull request closed without merging**